src/Wallabag/CoreBundle/Helper/EntriesExport.php

```diff
@@ -0,0 +1,263 @@
+        $content = new \MOBIFile();
+
+        /*
+         * Book metadata
+         */
+
+        $content->set('title', $this->title);
+        $content->set('author', implode($this->authors));
+        $content->set('subject', $this->title);
+
+        /*
+         * Front page
+         */
+
+        $content->appendParagraph('<div style="text-align:center;" ><p>'._('Produced by wallabag with PHPMobi').'</p><p>'._('Please open <a href="https://github.com/wallabag/wallabag/issues" >an issue</a> if you have trouble with the display of this E-Book on your device.').'</p></div>');
+        $content->appendImage(imagecreatefrompng('themes/_global/img/appicon/apple-touch-icon-152.png'));
+        $content->appendPageBreak();
+
+        /*
+         * Adding actual entries
+         */
+
+        foreach ($this->entries as $entry) {
+            $content->appendChapterTitle($entry->getTitle());
+            $content->appendParagraph($entry->getContent());
+            $content->appendPageBreak();
+        }
+        $mobi->setContentProvider($content);
+
+        // the browser inside Kindle Devices doesn't likes special caracters either, we limit to A-z/0-9
+        $this->title = preg_replace('/[^A-Za-z0-9\-]/', '', $this->title);
+
+        // we offer file to download
+        $mobi->download($this->title.'.mobi');
+    }
+
+    private function producePDF()
+    {
+        $pdf = new \TCPDF(PDF_PAGE_ORIENTATION, PDF_UNIT, PDF_PAGE_FORMAT, true, 'UTF-8', false);
+
+        /*
+         * Book metadata
+         */
+
+        $pdf->SetCreator(PDF_CREATOR);
+        $pdf->SetAuthor('wallabag');
+        $pdf->SetTitle($this->title);
+        $pdf->SetSubject('Articles via wallabag');
+        $pdf->SetKeywords('wallabag');
+
+        /*
+         * Front page
+         */
+
+        $pdf->AddPage();
+        $intro = '<h1>'.$this->title.'</h1><div style="text-align:center;" >
+        <p>'._('Produced by wallabag with tcpdf').'</p>
+        <p>'._('Please open <a href="https://github.com/wallabag/wallabag/issues" >an issue</a> if you have trouble with the display of this E-Book on your device.').'</p>
+        <img src="themes/_global/img/appicon/apple-touch-icon-152.png" /></div>';
+
+        $pdf->writeHTMLCell(0, 0, '', '', $intro, 0, 1, 0, true, '', true);
+
+        /*
+         * Adding actual entries
+         */
+
+        foreach ($this->entries as $entry) {
+            foreach ($this->tags as $tag) {
+                $pdf->SetKeywords($tag['value']);
+            }
+
+            $pdf->AddPage();
+            $html = '<h1>'.$entry->getTitle().'</h1>';
+            $html .= $entry->getContent();
+            $pdf->writeHTMLCell(0, 0, '', '', $html, 0, 1, 0, true, '', true);
+        }
+
+        // set image scale factor
+        $pdf->setImageScale(PDF_IMAGE_SCALE_RATIO);
+
+        $pdf->Output($this->title.'.pdf', 'D');
+    }
+
+    private function produceCSV()
+    {
+        header('Content-type: application/csv');
```

Comment:
You should better create your own Response (from the HttpFoundation) instead of using php internal way.